### PR TITLE
Refactor all command

### PR DESCRIPTION
**Pull request by avimanyum**

### CODE REVIEW

File: dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/subcommands/impl/Parent.java [hunk 1]
```diff
@@ -66,90 +67,21 @@ public void execute(final Namespace ns, DockerfileGitHubUtil dockerfileGitHubUti
...
-                            exceptions.add(e);
-                        }
-                    } else {
-                        log.warn("Didn't find fork for {} so not changing Dockerfiles", currUserRepo);
-                    }
+                try {
+                    commonSteps.prepareToCreatePullRequests(ns, pullRequestSender,
+                            contentsFoundWithImage.get(i), gitForkBranch, dockerfileGitHubUtil);
+                } catch (IOException e) {
+                    log.error("Could not send pull request.");
```
Comment (afalko) [inline dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/subcommands/impl/Parent.java:78]:
```suggestion
                    log.error("Could not send pull request.", e);
```

File: dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java [hunk 2]
```diff
@@ -437,4 +436,57 @@ public boolean thisUserIsOwner(GHRepository repo) throws IOException {
...
         return contentsWithImage;
     }
+
+    public void changeDockerfiles(Namespace ns,
+                                     Multimap<String, GitHubContentToProcess> pathToDockerfilesInParentRepo,
+                                     GitHubContentToProcess gitHubContentToProcess,
+                                     List<String> skippedRepos,
+                                     GitForkBranch gitForkBranch) throws IOException,
+            InterruptedException {
+        // Should we skip doing a getRepository just to fill in the parent value? We already know this to be the parent...
```
Comment (afalko) [inline dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java:446]: Good point; I wonder if that removes an API call

File: dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java [hunk 3]
```diff
@@ -437,4 +436,57 @@ public boolean thisUserIsOwner(GHRepository repo) throws IOException {
...
+        GHRepository parent = gitHubContentToProcess.getParent();
+        GHRepository forkedRepo = gitHubContentToProcess.getFork();
+        // TODO: Getting a null pointer here for someone... probably just fixed this since we have parent
```
Comment (afalko) [inline dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java:449]: "for someone"?

File: dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java [hunk 4]
```diff
@@ -437,4 +436,57 @@ public boolean thisUserIsOwner(GHRepository repo) throws IOException {
...
+            skippedRepos.add(forkedRepo.getFullName());
+        }
+
+        if (isContentModified) {
+            PullRequestInfo pullRequestInfo =
+                    new PullRequestInfo(ns.get(Constants.GIT_PR_TITLE),
+                            gitForkBranch.getImageName(),
+                            gitForkBranch.getImageTag(),
+                            ns.get(Constants.GIT_PR_BODY));
+            // TODO: get the new PR number and cross post over to old ones
```
Comment (afalko) [inline dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java:484]: What do you mean by cross post? Shouldn't we close the old PRs?

File: dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java [hunk 5]
```diff
@@ -437,4 +436,57 @@ public boolean thisUserIsOwner(GHRepository repo) throws IOException {
...
+            createPullReq(parent,
+                    gitForkBranch.getBranchName(),
+                    forkedRepo,
+                    pullRequestInfo);
+            // TODO: Run through PRs in fork to see if they have head branches that match the prefix and close those?
```
Comment (afalko) [inline dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java:489]: We either need to close and create new ones or update the existing ones.

**Comment by avimanyum** [comment]
@afalko I should have mentioned this earlier - all the comments you see here already existed in the code base. None of these were added as a part of this particular refactoring.

**Pull request merged**